stable_baselines/common/base_class.py

```diff
@@ -153,6 +155,49 @@ def _setup_learn(self, seed):
         if seed is not None:
             set_global_seeds(seed)
 
+    @abstractmethod
+    def _get_parameter_list(self):
+        """
+        Get tensorflow Variables of model's parameters
+
+        This includes all variables necessary for continuing training (saving / loading).
+
+        :return: (list) List of tensorflow Variables
+        """
+        pass
+
+    def get_parameters(self):
+        """
+        Get current model parameters as dictionary of variable name -> ndarray.
+
+        :return: (dict) Dictionary of variable name -> ndarray of model's parameters.
+        """
+        parameters = self._get_parameter_list()
+        parameter_values = self.sess.run(parameters)
+        return_dictionary = dict((param.name, value) for param, value in zip(parameters, parameter_values))
+        return return_dictionary
+
+    def _setup_load_operations(self):
+        """
+        Create tensorflow operations for loading model parameters
+        """
+        # Assume tensorflow graphs are static -> check
+        # that we only call this function once
+        if self._param_load_ops is not None:
+            raise RuntimeError("Parameter load operations have already been created")
+        # For each loadable parameter, create appropiate
+        # placeholder and an assign op, and store them to
+        # self.load_param_ops as dict of variable.name -> (placeholder, assign)
+        loadable_parameters = self._get_parameter_list()
+        # Use OrderedDict to store order for backwards compatability with
+        # list-based params
+        self._param_load_ops = OrderedDict()
+        with self.graph.as_default():
+            for param in loadable_parameters:
+                placeholder = tf.placeholder(dtype=param.dtype, shape=param.shape)
```

Comment:
I don't understand why you need to create those placeholders... I would expect the `param` to suffice.

Reply:
If you call `param.assign(ndarray)`, it creates a new op in the graph. By doing placeholders and assign ops on them, we avoid constantly creating ops.